README.md

```diff
@@ -1,121 +1,183 @@
-the SMC flash memory. This can be used to verify the update.
+## Bootloader Entry Points
 
-Finally, the update program must use command 0x82 to reset the SMC. This will
-write any remaining buffered data to flash memory and then turn off the computer.
+The bootloader has two entry points: the main entry point at address 0x1e00, and the start update entry point at address 0x1e02.
 
-The SMC can only update the flash memory in whole pages of 64 bytes. The committed
-bytes are buffered until there is a full page that can be written to flash memory.
-
-On receiving the first full page (address 0x0000-0x003f), the bootloader takes
-these special actions:
+### Main Entry Point (0x1e00)
 
-- Erasing the firmware area of the flash memory (address 0x0000-0x1dff)
+The SMC is always running even if the computer is turned off. As soon as you connect the system to mains power, the SMC executes its reset
+procedure. That ends by calling the reset vector at address 0x0000. When the bootloader is installed, the reset vector always jumps to
+the bootloader main entry point at address 0x1e00.
 
-- Moving the firmware's reset vector (address 0x0000) to EE_RDY (address 0x0012)
+The main entry checks if the reset button is being pressed. 
 
-- Pointing the reset vector to the bootloader main vector (address 0x1e00)
+If the button is pressed, the computer is turned on, and the SMC update procedure is started. This method of starting the bootloader, holding down the reset button while
+connecting the system to power, works in most situations even if the firmware has been bricked. Read more about that
+under [fail-safe](#fail-safe). The update program running on the Commander X16 must be loaded and started without keyboard or mouse support, 
+which are not available while the bootloader is running. One option is to store the update program as an autostarting program on the SD card.
 
-The reason for this is explained in the SMC reset process below.
+If the reset button is not pressed, execution continues with the firmware's own reset vector. That vector is moved to the EE_RDY (EEPROM Ready) vector
+at address 0x0012 when updating the firmware using the bootloader. The EE_RDY vector is not used by standard Arduino libraries, and is
+used for the same purpose by Optiboot. The solution prevents using the EEPROM Ready interrupt in firmware code.
 
+### Start Update Entry Point (0x1e02)
 
-## SMC Reset Process and Starting the Bootloader Recovery Update
+The start update entry point is to be called while the computer is running. The update
+procedure is started immediately after calling this entry point.
 
-The SMC is reset when first connecting it to power or when 
-the reset pin (#10) is grounded. On reset, execution always starts from the reset 
-vector (address 0x0000).
+A program running on the X16 cannot directly call the start update entry point. It must be done through
+a firmware command offset. Currently the firmware's command offset 0x8f does this.
```

Comment:
"firmware command offset" is, in my opinion, a complicated way of saying "i2c command" ("command" instead of "offset").

My suggestion: "To let the SMC jump to this entry point, the X16 program have to send the I2C command 0x8f (start bootloader)."